platform/persistence/platform-persistence-commands/src/main/java/org/codice/ddf/persistence/commands/AbstractStoreCommand.java

```diff
@@ -81,4 +85,30 @@ public Object execute() {
 
   /** Calls a command that operates on the Persistent Store service. */
   abstract void storeCommand() throws PersistenceException;
+
+  protected String createCql(String user, String cql) {
+    if (StringUtils.isBlank(user)) return "";
+    if (StringUtils.isNotBlank(cql)) {
+      cql = "( " + cql + ") AND user='" + user + "'";
+    } else {
+      cql = "user='" + user + "'";
+    }
+    return cql;
+  }
+
+  protected List<Map<String, Object>> getResults() throws PersistenceException {
+
+    List<Map<String, Object>> results = new ArrayList<>();
+    List<Map<String, Object>> pagedResults;
+    int startIndex = 0;
+    int pageSize = 1000;
+
+    do {
+      pagedResults = persistentStore.get(type, cql, startIndex, pageSize);
```

Comment:
:exclamation: We might have a problem with this call on SolrCloud since it looks like we do not sort to guarantee consistent ordering between shards.  

https://github.com/codice/ddf/blob/master/platform/persistence/platform-persistence-core-impl/src/main/java/org/codice/ddf/persistence/internal/PersistentStoreImpl.java#L191-L194

We have to sort by ID to avoid this issue with the catalog dump command

https://github.com/codice/ddf/blob/fc2fa5fb5f4226a408db48446d33238011a6a17b/catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DumpCommand.java#L253

Reply:
add default ID sorting on PersistentStore https://github.com/codice/ddf/pull/5745/files#diff-0927a922dc7dab1b936e3721db97721aR193